### PR TITLE
Use new Streams algorithm and mixins

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/streams/pull/1073.

***
<a href="https://whatpr.org/encoding/224.html" title="Last updated on Sep 15, 2020, 8:36 AM UTC (7bfd7db)">Preview</a> | <a href="https://whatpr.org/encoding/224/46711e6...7bfd7db.html" title="Last updated on Sep 15, 2020, 8:36 AM UTC (7bfd7db)">Diff</a>

### CODE REVIEW

File: encoding.bs [hunk 1]
```diff
@@ -1660,20 +1631,16 @@ constructor steps are:
      <li><p>Let <var>outputChunk</var> be the result of running <a>serialize stream</a> with
      <var>decoder</var> and <var>output</var>.
 
-     <li><p>if <var>outputChunk</var> is non-empty, call
-     <a abstract-op>TransformStreamDefaultControllerEnqueue</a>(<var>controller</var>,
-     <var>outputChunk</var>).
-
-     <li><p>Return a new promise resolved with undefined.
+     <li><p>If <var>outputChunk</var> is non-empty, then <a for=TransformStream>enqueue</a>
+     <var>outputChunk</var> in <var>decoder</var>'s <a for=GenericTransformStream>transform</a>.
```
Comment (andreubotella) [inline encoding.bs:1635]:
```suggestion
     <var>outputChunk</var> in <var>decoder</var>'s <a for=GenericTransformStream>transform</a>.

     <li><p>Return.
```

Needed to break the endless loop.

File: encoding.bs [hunk 2]
```diff
@@ -1610,19 +1587,18 @@ constructor steps are:
...
  <li><p>Let <var>flushAlgorithm</var> be an algorithm which takes no arguments and runs the
  <a>flush and enqueue</a> algorithm with <a>this</a>.
 
- <li><p>Let <var>transform</var> be the result of calling
- <a abstract-op>CreateTransformStream</a>(<var>startAlgorithm</var>, <var>transformAlgorithm</var>,
- <var>flushAlgorithm</var>).
-
- <li><p>Set <a>this</a>'s <a for=GenericTransformStream>transform</a> to <var>transform</var>.
+ <li><p>Set <a>this</a>'s <a for=GenericTransformStream>transform</a> to the result of
+ <a for=TransformStream>creating</a> a {{TransformStream}} with
```
Comment (annevk) [inline encoding.bs:1597]:
```suggestion
 <a for=TransformStream>creating</a> a {{TransformStream}} object with
```

File: encoding.bs [hunk 3]
```diff
@@ -1660,20 +1631,18 @@ constructor steps are:
...
     </ol>
 
    <li><p>Let <var>result</var> be the result of <a>processing</a> <var>token</var> for
    <var>decoder</var>'s <a for=TextDecoderCommon>decoder</a>, <var>decoder</var>'s
    <a for=TextDecoderCommon>stream</a>, <var>output</var>, and <var>decoder</var>'s
    <a for=TextDecoderCommon>error mode</a>.
 
-   <li><p>If <var>result</var> is <a>error</a>, then return a new promise rejected with a
-   {{TypeError}} exception.
+   <li><p>If <var>result</var> is <a>error</a>, then throw a {{TypeError}} exception.
```
Comment (annevk) [inline encoding.bs:1645]: Should we drop the trailing " exception" here as we do in other standards?

**Review by annevk — APPROVED** [review]
I really like leaning into IDL more heavily for doing the promise wrapping. I should do that more often.

File: encoding.bs (hunk 2 above)
Comment (domenic) [reply encoding.bs:1597]: This mismatches the algorithm declaration, which does not include "object".

File: encoding.bs (hunk 3 above)
Comment (domenic) [reply encoding.bs:1645]: I don't have an opinion on this one.

Comment (annevk) [reply encoding.bs:1645]: Let's drop it then, I can make this change as well before landing.

**Comment by annevk** [comment]
I rebased this and pushed some nits and corrected things that seemed wrong. Still needs upstream changes to merge.

**Review by ricea — COMMENTED** [review]
lgtm

**Comment by domenic** [comment]
Thank you for catching those; your changes LGTM. Will work on getting upstream working now.

**Comment by domenic** [comment]
The Streams PR is merged; once the link databases update let's re-run CI and see if this works.

**Pull request merged**